pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -794,6 +803,41 @@ if [ -z "$rollback" ]; then
         else
             pathToConfig="$(nixFlakeBuild "$flake#$flakeAttr.config.system.build.vmWithBootLoader" "${extraBuildFlags[@]}" "${lockFlags[@]}")"
         fi
+    elif [ "$action" = build-image ]; then
+        if [[ -z $buildingAttribute ]]; then
+            variants="$(
+                runCmd nix-instantiate --eval --json --expr \
+                "let
+                    value = import \"$(realpath $buildFile)\";
+                    set = if builtins.isFunction value then value {} else value;
```

Comment:
Not proud of this and happy for hints on how to improve this. It's meant to support the case where e.g. `default.nix` returns a function where each parameter has a default value, i.e. the common `{pkgs ? import <nixpkgs> {}}:`. nix-build seems to handle it fine, but nix-instantiate does not.

Reply:
Note to self: Set key, like lib.applyModuleArgsIfFunction does.